pkg/assembler/backends/inmem/certifyBad.go

```diff
@@ -282,14 +286,13 @@ func (c *demoClient) addCBIfMatch(out []*model.CertifyBad,
 	filter *model.CertifyBadSpec, link *badLink) (
 	[]*model.CertifyBad, error) {
 
-	if filter != nil && noMatch(filter.Justification, link.justification) {
-		return out, nil
-	}
-	if filter != nil && noMatch(filter.Collector, link.collector) {
-		return out, nil
-	}
-	if filter != nil && noMatch(filter.Origin, link.origin) {
-		return out, nil
+	if filter != nil {
+		if noMatch(filter.Justification, link.justification) ||
+			noMatch(filter.Collector, link.collector) ||
+			noMatch(filter.Origin, link.origin) ||
+			(filter.KnownSince != nil && filter.KnownSince.After(link.knownSince)) {
```

Comment:
Should be:
`&& !filter.KnownSince.Equal(link.knownSince)`
right?

Reply:
This is to include anything before a timestamp specified by a user in the filter. This is similar to other verbs. So anything `After` would be omitted.

Reply:
Ah, interesting. We should probably update the comments on the spec.

Reply:
@jeffmendoza I don't understand what spec you mean.

Reply:
I have updated the specs for hasSBOM, certifyBad, and certifyGood.